httpclient/config.go

```diff
@@ -0,0 +1,400 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"errors"
+	"github.com/acronis/go-appkit/config"
+	"github.com/acronis/go-appkit/retry"
+	"github.com/cenkalti/backoff/v4"
+	"time"
+)
+
+const (
+	// DefaultClientWaitTimeout is a default timeout for a client to wait for a request.
+	DefaultClientWaitTimeout = 10 * time.Second
+
+	// RetryPolicyExponential is a policy for exponential retries.
+	RetryPolicyExponential = "exponential"
+
+	// RetryPolicyConstant is a policy for constant retries.
+	RetryPolicyConstant = "constant"
+
+	// configuration properties
+	cfgKeyRetriesEnabled                          = "retries.enabled"
+	cfgKeyRetriesMax                              = "retries.maxAttempts"
+	cfgKeyRetriesPolicyStrategy                   = "retries.policy.strategy"
+	cfgKeyRetriesPolicyExponentialInitialInterval = "retries.policy.exponentialBackoffInitialInterval"
+	cfgKeyRetriesPolicyExponentialMultiplier      = "retries.policy.exponentialBackoffMultiplier"
+	cfgKeyRetriesPolicyConstantInternal           = "retries.policy.constantBackoffInterval"
+	cfgKeyRateLimitsEnabled                       = "rateLimits.enabled"
+	cfgKeyRateLimitsLimit                         = "rateLimits.limit"
+	cfgKeyRateLimitsBurst                         = "rateLimits.burst"
+	cfgKeyRateLimitsWaitTimeout                   = "rateLimits.waitTimeout"
+	cfgKeyLoggerEnabled                           = "logger.enabled"
```

Comment:
What do you think about renaming logger to just log? This would improve consistency with the `httpserver.log` configuration (https://github.com/acronis/go-appkit/blob/64b2af8edd00fc94e251c12fcfe6512f4ffa36d4/httpserver/config.go#L28C41-L28C51).